topic4/contracts/Math.sol

```diff
@@ -3,5 +3,48 @@ pragma solidity ^0.4.23;
 import './interfaces/IMath.sol';
 
 contract Math is IMath {
-		
+
+    uint _stateNumber = 0;
+
+	function stateNumber() external view returns (uint) {
+        return _stateNumber;
+    }
+
+	function resetStateNumber() external {
+        _stateNumber = 0;
+    }
+
+	function add(uint256 _inputNumber) external {
+        uint sum = _stateNumber + _inputNumber;
+		assert(sum >= _stateNumber);
+		_stateNumber = sum;
+    }
+
+	function sub(uint256 _inputNumber) external {
+		assert(_inputNumber <= _stateNumber);
+        uint diff = _stateNumber - _inputNumber;
```

Comment:
5. Since the logic is easy to follow, you may save a little gas by not getting this `diff` variable into the memory, so it becomes `_stateNumber -= _inputNumber;`.  However in more complex scenarios do not sacrifice the readability.